### PR TITLE
Support for @JsonDeserialize(builder=..)

### PR DESCRIPTION
**Pull request by graemerocher**
Draft PR for builder support, some open questions:

* Don't know how to support pre-instantiation hook
* Unsure about strict nullable

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,4 +5,5 @@ plugins {
 
 repositories {
     mavenCentral()
+    mavenLocal()
```
Comment (yawkat) [inline build.gradle.kts:8]: remove when this pr is ready

**Review by yawkat — APPROVED** [review]

**Comment by yawkat** [comment]
what exactly are the pre-instantiate hooks used for (theyre used in data right)?

i think builders should have the same behavior as creator methods wrt pre-instantiate hooks. i think atm the hook is called after the creator but before any more properties are set.

**Comment by graemerocher** [comment]
I am not sure about these pre-instantiate hooks myself, @dstepanov will have to confirm

**Comment by dstepanov** [comment]
The hook is used to perform possible validation before the constructor is invoked. In this case, we need to think about how we would trigger the builder method validation.

**Pull request merged**